packages/xo-server/src/xo-mixins/proxies.js

```diff
@@ -60,48 +69,168 @@ export default class Proxy {
     return this._db.remove(id)
   }
 
-  async getProxy(id) {
+  async destroyProxy(id) {
+    const { vmUuid } = await this._getProxy(id)
+    if (vmUuid !== undefined) {
+      await this._app.getXapi(vmUuid).deleteVm(vmUuid)
+    }
+    return this.unregisterProxy(id)
+  }
+
+  async _getProxy(id) {
     const proxy = await this._db.first(id).then(extractProperties)
     if (proxy === undefined) {
       throw noSuchObject(id, 'proxy')
     }
     return proxy
   }
 
+  getProxy(id) {
+    return this._getProxy(id).then(omitToken)
+  }
+
   getAllProxies() {
-    return this._db.get()
+    return this._db.get().then(proxies => proxies.map(omitToken))
   }
 
   @synchronizedWrite
-  async updateProxy(id, { address, authenticationToken, name }) {
-    const proxy = await this.getProxy(id)
-    if (address !== undefined && proxy.address !== address) {
-      await this._throwIfRegistered(address)
-    }
+  async updateProxy(id, { address, authenticationToken, name, vmUuid }) {
+    await this._throwIfRegistered(address, vmUuid)
```

Comment:
What if we're not updating the `address` nor the `vmUuid`?

Reply:
It will not throw.

Reply:
Even if they're not `undefined`? I mean if we call `updateProxy` with the same address that is already registered for that proxy, for example.